src/poetry/utils/cache.py

```diff
@@ -172,7 +184,12 @@ def _get_payload(self, key: str) -> T | None:
             return None
 
         with open(path, "rb") as f:
-            payload = self._deserialize(f.read())
+            file_content = f.read()
+
+        try:
+            payload = self._deserialize(file_content)
```

Comment:
Thinking about this again.. I wonder if we can skip the custom error altogether and simply run the forget+return `None` routine here. `_get_payload` already manages forgetting in case the payload is expired. I'm not too opinionated about this, though.

Reply:
I suppose it is a little extra. Removed.